lib/SILOptimizer/Utils/InstOptUtils.cpp

```diff
@@ -753,29 +770,42 @@ getConcreteValueOfExistentialBoxAddr(SILValue addr, SILInstruction *ignoreUser)
   for (Operand *stackUse : stackLoc->getUses()) {
     SILInstruction *stackUser = stackUse->getUser();
     switch (stackUser->getKind()) {
-      case SILInstructionKind::DeallocStackInst:
-      case SILInstructionKind::DebugValueAddrInst:
-      case SILInstructionKind::LoadInst:
-        break;
-      case SILInstructionKind::StoreInst: {
-        auto *store = cast<StoreInst>(stackUser);
-        assert(store->getSrc() != stackLoc && "cannot store an address");
-        // Bail if there are multiple stores.
-        if (singleStackStore)
-          return SILValue();
-        singleStackStore = store;
-        break;
-      }
-      default:
-        if (stackUser != ignoreUser)
+    case SILInstructionKind::DestroyAddrInst: {
+      // Make sure the destroy_addr is the instruction before one of our
+      // dealloc_stack insts and is directly on the stack location.
+      auto next = std::next(stackUser->getIterator());
+      if (auto *dsi = dyn_cast<DeallocStackInst>(next))
+        if (dsi->getOperand() != stackLoc)
```

Comment:
This `destroy_addr` check is new, super-specific, brittle, and bizarre. Can you at least explain in the comment why you need it.